doc/mgmt/config_replace_rollback_HLD.md

```diff
@@ -0,0 +1,415 @@
+| **Example**                                             | Sonic#copy http://inter128.acme.com/sample.json running-configuration replace<br>Sonic# |
+| **Support**                                             | 4.1 |
+
+#### 3.2.2.2 Configuration Session Mode Commands
+##### 3.2.2.2.1 Replace Command
+| replace command                                         | replace command inside configuration session to replace running configuration  |
+| ------------------------------------------------------- | ------------------------------------------------------------ |
+| **Syntax**                                              | replace source-url  |
+| **Parameters**                                          |
+| source-url                                              | config://filename<br>home://filename<br>ftp://[userid[:passwd]@]{hostname \| host-ip}/file-path<br>http[s]://[userid[:passwd]@]{hostname \| host-ip}/file-path<br>scp://[userid[:passwd]@]{hostname \| host-ip}/file-path|
+| **Default**                                             |  |
+| **Command mode**                                        | Config-s |
+| **Roles**                                               | admin |
+| **Example**                                             | Sonic(config-s)#replace http://inter128.acme.com/sample.json<br>Sonic(config-s)# |
+| **Support**                                             | 4.1 |
+##### 3.2.2.2.2 commit command
+| commit command                                          | commit command inside configuration session to kick start the timer or confirm the configuration changes |
+| ------------------------------------------------------- | ------------------------------------------------------------ |
+| **Syntax**                                              | commit [label \<LABEL\>] [time seconds] \| [confirm]  |
+| **Parameters**                                          |
+| confirm						  | To make the configuration changes permanent in running-configuration |
+| time   						  | 30 .. 3600 seconds for which the configuration changes kept in running-configuration as trial stage, default - 30 |
+| label   						  | commit label string, upto 72 characters |
+| **Default**                                             | None |
+| **Command mode**                                        | Config-s |
+| **Roles**                                               | admin |
+| **Example**                                             | Sonic(config-s)#commit time 120<br>Sonic(config-s)#commit confirm |
+| **Support**                                             | 4.1 |
+Note: "commit" will be implemented by Broadcom. time and confirm options will be added to "commit" command
+### 3.2.3 REST API Support
+Regular RPC operation can be sent from REST. The server side implementation should be the same as exce mode CLI.
+
+### 3.2.4 gNMI Support
+
+
+# 4 Flow Diagrams
+# 4.1 Exec mode configuration replace command
```

Comment:
There is no explanation on how diff is calculated, how to detect there is an unsupported configuration.

Reply:
Map based comparison between running and intended config is optimally performed. Unsupported configurations like DPB etc. will be maintained and checked from the diff data set.